dicom_utils/volume.py

```diff
@@ -128,8 +128,9 @@ def __repr__(self) -> str:
         return s
 
     def get_indices(self, total_frames: Optional[int]) -> Tuple[int, Optional[int], int]:
```

Comment:
If we make that change, we will need to assume that a total frame count is always available from the DICOM. The case of `total_frames=None` arises when `dcm.NumberOfFrames` is not present.

Reply:
Agreed, we shouldn't assume that. The way this function is written, it assumes that `total_frames` is not _None_. So, I think this function should either handle the _None_ case, or we should change the type hint to just `int` so that the user knows to handle the _None_ case before calling this function.
Also, not sure if you saw that I also suggested the return type hint be changed to `Tuple[int, int, int]`.

Reply:
How about something like this?

```python
def slice_dicom(self, dcm: U) -> U:
    num_frames: Optional[SupportsInt] = dcm.get("NumberOfFrames", None)
    if num_frames is None and self.center is None:
        raise AttributeError(
            f"`NumberOfFrames` cannot be absent when `{self.__class__.__name__}.center` is `None`"
        )
    return super().slice_dicom(dcm)

def get_indices(self, total_frames: Optional[int]) -> Tuple[int, int, int]:
    if self.center is None and total_frames is None:
        raise ValueError(
            f"`total_frames` cannot be `None` when `{self.__class__.__name__}.center` is `None`"
        )
    center = self.center if self.center is not None else total_frames // 2
    start = center - self.before
    end = center + self.after + 1
    return start, end, self.stride
```

Reply:
Awesome, that looks like a good solution